core/src/scene/sceneLoader.cpp

```diff
@@ -64,6 +64,56 @@ void printFilters(const SceneLayer& layer, int indent){
     }
 };
 
+void SceneLoader::applyGlobalProperties(Node& node, Scene& scene) {
+    switch(node.Type()) {
+    case NodeType::Scalar:
+        {
+            std::string key = node.Scalar();
+            if (key.compare(0, 7, "global.") == 0) {
+                key.replace(0, 7, "");
+                key = std::regex_replace(key, std::regex("\\."), DELIMITER);
+                node = scene.globals()[key];
+            }
+        }
+        break;
+    case NodeType::Sequence:
+        for (const auto& n : node) {
+            Node nextNode = n;
+            applyGlobalProperties(nextNode, scene);
+        }
+        break;
+    case NodeType::Map:
+        for (const auto& n : node) {
+            Node nextNode = n.second;
+            applyGlobalProperties(nextNode, scene);
+        }
+        break;
+    default:
+        break;
+    }
+}
+
+void SceneLoader::parseGlobals(const Node& node, Scene& scene, const std::string& key) {
+    switch (node.Type()) {
+    case NodeType::Scalar:
+    case NodeType::Sequence:
+        {
+            auto& g = scene.globals();
+            g[key] = YAML::Clone(node);
+            break;
+        }
+    case NodeType::Map:
+        for (const auto& g : node) {
+            std::string value = g.first.Scalar();
+            Node global = node[value];
+            std::string mapKey = (key.size() == 0) ? value : (key + DELIMITER + value);
+            parseGlobals(global, scene, mapKey);
+        }
```

Comment:
For map nodes we need to create globals for not only the leaf nodes, but for all of the mappings themselves. For example, with the global block you've added for demonstration, it's valid for the scene to reference `global.colors` but there is no entry for it in `scene.globals()`